_data-prepper/pipelines/configuration/processors/otel-metrics-raw.md

```diff
@@ -0,0 +1,147 @@
+        exponential_histogram_max_allowed_scale: 10
+        flatten_attributes: false
+```
+
+Each array element describes one bucket. Each bucket contains the lower boundary, upper boundary, and its value count. This is a specific form of more detailed OpenTelemetry representation that is a part of the JSON output created by the following plugin. See the following `JSON` file, which is added to each histogram `JSON` by the OTel metrics raw processor:
+
+```json
+ "explicitBounds": [
+    5.0,
+    10.0
+  ],
+   "bucketCountsList": [
+    2,
+    5
+  ]
+```
+
+
+
+### `calculate_exponential_histogram_buckets`
+
+If `calculate_exponential_histogram_buckets` is set to `true` (the default setting), the following JSON values are added to each JSON histogram:
+
+```json
+
+    "negativeBuckets": [
+        {
+        "min": 0.0,
+        "max": 5.0,
+        "count": 2
+        },
+        {
+        "min": 5.0,
+        "max": 10.0,
+        "count": 5
+        }
+    ],
+...
+    "positiveBuckets": [
+        {
+        "min": 0.0,
+        "max": 5.0,
+        "count": 2
+        },
+        {
+        "min": 5.0,
+        "max": 10.0,
+        "count": 5
+        }
+    ],
+```
+
+The following JSON file is a more detailed form of the dense OpenTelemetry representation, which consists of negative and positive buckets, a scale parameter, offset, and list of bucket counts. 
+
+
+```json
+    "negative": [
+        1,
+        2,
+        3
+    ],
+    "positive": [
+        1,
+        2,
+        3
+    ],
+    "scale" : -3,
+    "negativeOffset" : 0,
+    "positiveOffset" : 1
+```
+
+
+### `exponential_histogram_max_allowed_scale`
+
+The `exponential_histogram_max_allowed_scale` parameter defines the maximum allowed scale for an exponential histogram. If you increase this parameter, you will increase potential memory consumption. See the [OpenTelemetry specifications](https://github.com/open-telemetry/opentelemetry-proto/blob/main/opentelemetry/proto/metrics/v1/metrics.proto) for more information on exponential histograms and their computational complexity.
+
+All exponential histograms that have a scale that is above the configured parameter (by default, a value of `10`) are discarded and logged with an error level. You can check the llog that Data Prepper creates to see the `ERROR` log message.
+
+**Note**: The absolute scale value is used for comparison, so a scale of `-11` that is treated equally to `11` exceeds the configured value of `10` and can be discarded.
+
+## Metrics
+
+The following table describes metrics that are common to all processors.
+
+| Metric name | Type | Description |
+| ------------- | ---- | -----------|
+| `recordsIn` | Counter | Metric representing the ingress of records to a pipeline component. |
+| `recordsOut` | Counter | Metric representing the egress of records from a pipeline component. |
+| `timeElapsed` | Timer | Metric representing the time elapsed during execution of a pipeline component. |
```

Comment:
@dlvenable Do you still want to keep this section?

Reply:
Yes, from the discussion we had offline we decided to keep these.

Reply:
pipeline component in the description is too generic. We should either replace it with processor name (which feels redundant to me since these metrics are documented for each processor separately). Or remove the pipeline component completely,  something similar to what I added below
```
| Metric name | Type | Description |
| ------------- | ---- | -----------|
| `recordsIn` | Counter | Metric representing the number of ingress records. |
| `recordsOut` | Counter | Metric representing the number of egress records. |
| `timeElapsed` | Timer | Metric representing the time elapsed during execution of records. |
```

Reply:
@asifsmohammed This looks great. Thank you!